ubuntu/entrypoint.sh

```diff
@@ -80,6 +84,84 @@ function unload_modules() {
     done
 }
 
+
+# find all available vfs
+# return format <pf_name> <vf_name> <vf_index> <vf_mac> <vf_pci_addr>
+function find_mlx_vfs() {
+    for ethpath in /sys/class/net/*; do
+        if (grep $VENDOR "$ethpath"/device/vendor >/dev/null 2>&1); then
+            for vf in "$ethpath"/device/virtfn[0-9]*/net/*; do
+                [[ -d $vf ]] && [[ -e $vf/address ]] && [[ -L $vf/device ]] || continue
+                pf_name=$(basename $ethpath) && [[ -n $pf_name ]] || return 1
+                vf_name=$(basename $vf) && [[ -n $vf_name ]] || return 1
+                vf_index=$(sed -E 's|.*/virtfn([0-9]+)/.*|\1|' <<<$vf) && [[ -n $vf_index ]] || return 1
+                vf_mac=$(cat $vf/address) && [[ -n $vf_mac ]] || return 1
+                vf_pci_addr=$(basename $(readlink $vf/device)) && [[ -n $vf_pci_addr ]] || return 1
+
+                echo "$pf_name" "$vf_name" "$vf_index" "$vf_mac" "$vf_pci_addr"
+            done
+        fi
+    done
+}
+
+function set_administrative_mac_for_vf() {
+    ip link set dev "$1" vf "$2" mac "$3"
+}
+
+function set_link_up() {
+    ip link set dev "$1" up
+}
+
+function driver_bind_vf() {
+    echo "$1" >$DRIVER_PATH/bind
+}
+
+function driver_unbind_vf() {
+    echo "$1" >$DRIVER_PATH/unbind
+}
+
+function fix_guid_for_vfs() {
+    local vf_data
+    vf_data="$(find_mlx_vfs)"
+    if [[ $? -ne 0 ]];then
+        echo "Failed to read info about VFs"
+        return 1
+    fi
+    declare -A pf_arr 
+    while read -r d; do
+        [[ -n "$d" ]] || continue
+        local vf_info=($d)
+        local pf_name=${vf_info[0]}
+        pf_arr[$pf_name]=1
+        local vf_name=${vf_info[1]}
+        local vf_index=${vf_info[2]}
+        local vf_mac=${vf_info[3]}
+        local vf_pci_addr=${vf_info[4]}
+
+        echo "fix guid for VF: $vf_name"
+
+        if ! set_administrative_mac_for_vf "$pf_name" "$vf_index" "$vf_mac"; then
+            echo "failed to set mac for  VF $vf_index"
```

Comment:
you have double space also you should specify the $pf_name

Reply:
fixed